### ISSUE TITLE
Can't run `repo bootstrap` with Bitbucket Server

### ISSUE THREAD

**Issue by consal** [argoproj-labs/argocd-autopilot#473]
Our company has an on-prem Bitbucket deployment. I created an HTTP token in my Bitbucket user profile with the necessary permissions and exported it as `GIT_TOKEN`
![image](https://github.com/argoproj-labs/argocd-autopilot/assets/51994060/e89924cb-64e9-4f66-94e8-bb69a2637632)

When running the command, I get the error `authentication required`.

```
export GIT_REPO="https://bitbucket.company.com/scm/~myuser/argocd-autopilot.git"
argocd-autopilot repo bootstrap --provider bitbucket-server
INFO cloning repo: https://bitbucket.company.com/scm/~myuser/argocd-autopilot.git
FATAL authentication required
```

Things I tried:

- Adding `--provider bitbucket-server` or `--provider bitbucket`
- Adding `--git-user myuser`
- Using ssh git repo URL
- Adding `log-level debug` doesn't show the actual command that is used to clone the repo

Without success. Can it work with an on-prem bitbucket?

**Comment by jinxmcg** [comment]
You need to create a app password not access token.
Go to: https://bitbucket.org/account/settings/ and copy USERNAME
then go to: https://bitbucket.org/account/settings/app-passwords/ and create a app password

```
export GIT_TOKEN="THE APP PASSWORD TOKEN"
export GIT_REPO="https://bitbucket.org/YOURUSER/YOUR_REPO.git"
export GIT_USER="USERNAME"

argocd-autopilot repo bootstrap --provider bitbucket
```

**Comment by todaywasawesome** [comment]
Yes `bitbucket-server` would be the right one to use. https://argocd-autopilot.readthedocs.io/en/stable/commands/argocd-autopilot_repo_bootstrap/#options

@consal were you able to get it working?

**Comment by jblsk** [comment]
Unfortunately the suggested endpoint doesn't exist in **bitbuckt-server**:
<img width="506" alt="image" src="https://github.com/user-attachments/assets/2b8c71eb-732c-4a31-ae58-2f8f97055395">
